File: 2015/go/day01/main.go
package main

import (
	"bufio"
	"log"
	"os"
)

func first(scanner *bufio.Scanner) {
	scanner.Scan()
	s := scanner.Text()

	floor := 0
	for _, v := range s {
		if string(v) == "(" {
			floor++
		} else {
			floor--
		}
	}
	log.Println(floor)
}

func second(scanner *bufio.Scanner) {
	scanner.Scan()
	s := scanner.Text()

	floor := 0
	position := 0
	for _, v := range s {
		if floor == -1 {
			log.Println(position)
			break
		}
		if string(v) == "(" {
			floor++
		} else {
			floor--
		}

		position++

	}
}

func main() {
	file, err := os.Open("input.txt")

	if err != nil {
		log.Fatalf("Can not open file")

	}

	defer file.Close()

	scanner := bufio.NewScanner(file)
	//first(scanner)
	second(scanner)

}

File: 2015/rust/day01/src/main.rs
use std::fs;

fn first(contents: String) {
    let mut floor = 0;
    for c in contents.chars() {
        if c == '(' {
            floor += 1;
        } else {
            floor -= 1;
        }
    }

    println!("{}", floor);
}

fn second(contents: String) {
    let mut floor = 0;
    let mut position = 0;
    for c in contents.chars() {
        if floor == -1 {
            println!("{}", position);
            break;
        }
        if c == '(' {
            floor += 1;
        } else {
            floor -= 1;
        }

        position += 1
    }

    println!("{}", floor);
}

fn main() {
    let contents = fs::read_to_string("./input.txt").expect("Error reading from file");
    //first(contents)
    second(contents)
}
